add Ab:RBD analysis scripts (#3)
choderalab/rbd-ab-contact-analysis: 02_analysis_scripts/README.md

## Analysis

> Below are exmaples of how to use analysis scripts to analyse the dataset for F@H data in [DOI link]. It is recommened to run each of these scripts in separate directories.

Reviewer: ```suggestion
Below are examples of how to use analysis scripts to analyse the dataset for F@H data in [DOI link]. It is recommended to run each of these scripts in separate directories.
```

**Note**: these scripts were developed as part of a time sensive project, as a result they are not memory efficient.

### Contact analysis

> `analyze_antibody_rbd_contacts.py` can be used to generate a series of `.pkl` files containing an N x M x No. frames array of distances between heavy atoms at the RBD:Ab interface.

Reviewer: can you define N and M here?

> For example, analysing the F@H `17341` project (S2H97:Ab):

Reviewer: ```suggestion
For example, analysing the F@H `17347` project (RBD:S309):
```

Reviewer: Can you add a line above this that says something like: 
Project numbers:
RBD:S309 - 17341
RBD:S2H97 - 17347

```python

> python analyze_antibody_rbd_contacts.py -mutant WT -pdb_file system.pdb -traj_path path/to/traj/ -fah_project_code 17341 -chain_info_file chain_info_S2H97_RBD.yaml -run_number 0 -discard_time 100

Reviewer: can you define what `-chain_info_file` and `-run_number` and `-discard_time` correspond to?

Reviewer: And also specify that there are 5 RUNs for each project, so we ran this command on each of the 5 RUNs separately. And that we used a discard time of 100 for all analysis jobs.

> python analyze_antibody_rbd_contacts.py -mutant WT -pdb_file system.pdb -traj_path path/to/traj/ -fah_project_code 17341 -chain_info_file chain_info_S2H97_RBD.yaml -run_number 0 -discard_time 100

Reviewer: ```suggestion
python analyze_antibody_rbd_contacts.py -mutant WT -pdb_file system.pdb -traj_path path/to/traj/ -fah_project_code 17341 -chain_info_file chain_info_S309_RBD.yaml -run_number 0 -discard_time 100
```

```

> * `system.pdb` should contain no solvent.

Reviewer: ```suggestion
* `system.pdb` is a PDB of the RBD:antibody without waters, but with ions.
```

### Interaction count
The `.pkl` files generated from `analyze_antibody_rbd_contacts.py` can then be used with the `count_ab_rbd_interactions.py` script. Since the Ab typically contains two chains, the antibody chain number needs to specified.

```python
python count_ab_rbd_interactions.py -path_name path/to/contacts/ -ab_chain_key 1 -metadata_file PROJ17341_WT_selection_metadata.pkl -project_number 17341 -mutant WT
```

* `path/to/contacts/` should be a path containing the `.pkl` files generated by `analyze_antibody_rbd_contacts.py`.
* The `-metadata` flag should point to a `.pkl` metadata file generated by `analyze_antibody_rbd_contacts.py`.

### Plotting
Running `count_ab_rbd_interactions.py` produces one `.pkl` file per antibody chain (e.g. count_chain1.pkl and count_chain2.pkl). These can then be used in the `make_hmaps.py` script.

```python

> python make_hmaps.py -data count_chain1.pkl count_chain2.pkl -save_path path/to/save/plots -ab_name S2H97 -custom_residues_file s2h97_ab_custom_residues.yaml

Reviewer: can you define what `-custom_residues_file` corresponds to?

> python make_hmaps.py -data count_chain1.pkl count_chain2.pkl -save_path path/to/save/plots -ab_name S2H97 -custom_residues_file s2h97_ab_custom_residues.yaml

Reviewer: ```suggestion
python make_hmaps.py -data count_chain1.pkl count_chain2.pkl -save_path path/to/save/plots -ab_name S2H97 -custom_residues_file s309_ab_custom_residues.yaml
```

For example, analysing the F@H `17347` project (S2H97:Ab):

```python
python analyze_antibody_rbd_contacts.py -mutant WT -pdb_file system.pdb -traj_path path/to/traj/ -fah_project_code 17347 -chain_info_file chain_info_S2H97_RBD.yaml -run_number 0 -discard_time 100
```
* `system.pdb` is a PDB of the RBD:antibody system without waters, but with Na+ and Cl- ions.
* `path/to/traj/` should be a path to munged F@H data (Data available at: [S2H97](https://covid.molssi.org//simulations/#foldinghome-simulations-of-the-sars-cov-2-spike-rbd-bound-to-monoclonal-antibody-s2h97), [S2H97](https://covid.molssi.org//simulations/#foldinghome-simulations-of-the-sars-cov-2-spike-rbd-bound-to-monoclonal-antibody-S2H97)).
* The `-chain_info_file` flag should point to a YAML file in the format:
```
antibody_chains:
  chain_1: A
  chain_2: B
rbd_chain:
  chain_1: R
glycan_chain:
  chain_1: X
```
where `A`, `B`, `R`, and `X` are the names of the indicated chain in the PDB file.

> * The `-discard_time` flag specifies how much time to discard for a particular CLONE before starting contact analysis.

Reviewer: ```suggestion
* The `-discard_time` flag specifies how much time (ns) to discard for a particular CLONE before starting contact analysis.
```

> For this project there were a total of 5 RUNs, so the above script was ran on each RUN separately. Additioanlly a `-discard_time` of 100 ns was applied

Reviewer: ```suggestion
For this project there were a total of 5 RUNs, so the above script was ran on each RUN separately. Additionally a `-discard_time` of 100 ns was applied
```

```

* `path/to/contacts/` should be a path containing the `.pkl` files generated by `analyze_antibody_rbd_contacts.py`.
* The `-metadata` flag should point to a `.pkl` metadata file generated by `analyze_antibody_rbd_contacts.py`.

For this project there were a total of 5 RUNs, the `.pkl` count files were combined using `combine_dataframes.py`, and exmaple is given below:

```python
python combine_dataframes.py -path path/to/count/files/ -run_number 0 -project_number 17347
```

* `path/to/count/files/` should contain the directory containing the count `.pkl` files from `count_ab_rbd_interactions.py`.
* `-run_number` corresponds to the F@H RUN number.
* `-project_number` corresponds to the F@H project number.

### Plotting
Running `count_ab_rbd_interactions.py` produces one `.pkl` file per antibody chain (e.g. count_chain1.pkl and count_chain2.pkl). These can then be used in the `make_hmaps.py` script.

```python
python make_hmaps.py -data count_chain1.pkl count_chain2.pkl -save_path path/to/save/plots -ab_name S2H97 -custom_residues_file s2h97_ab_custom_residues.yaml
```

> * The `-custom_residues_file` should point to a YAML file containing the system type (e.g. WT). See `s2h97_ab_custom_residues.yaml` for an example.

Reviewer: I think you mean:
```suggestion
* The `-custom_residues_file` should point to a YAML file containing the RBD residues for which to compute the % contribution to total interface close contacts. See `s2h97_ab_custom_residues.yaml` for an example.
```
